src/Sulu/Bundle/ContentBundle/Command/WebspaceCopyCommand.php

```diff
@@ -0,0 +1,667 @@
+     *
+     * @param string $localeSource
+     * @param string $localeDestination
+     */
+    protected function copyRedirectsAndStructure(
+        $localeSource,
+        $localeDestination
+    ) {
+        $this->output->writeln([
+            '------------------------------',
+            '<info>Webspace: </info>' . $this->webspaceKeySource . ' => ' . $this->webspaceKeyDestination,
+            '<info>Locale: </info>' . $localeSource . ' => ' . $localeDestination,
+            '------------------------------',
+        ]);
+
+        /** @var HomeDocument $homeDocumentSource */
+        $homeDocumentSource = $this->documentManager->find(
+            $this->sessionManager->getContentPath($this->webspaceKeySource),
+            $localeSource
+        );
+
+        // Generate redirect and structure.
+        $this->recursiveCopyRedirectsAndStructure(
+            $homeDocumentSource,
+            $localeDestination
+        );
+
+        $this->output->writeln([
+            '------------------------------',
+            '',
+        ]);
+    }
+
+    /**
+     * @param BasePageDocument $documentSource
+     * @param BasePageDocument|null $parentDocumentDestination
+     * @param string $localeDestination
+     */
+    protected function recursiveCopy(
+        BasePageDocument $documentSource,
+        BasePageDocument $parentDocumentDestination = null,
+        $localeDestination
+    ) {
+        // Generate new path.
+        $newPath = str_replace(
+            $this->sessionManager->getContentPath($this->webspaceKeySource),
+            $this->sessionManager->getContentPath($this->webspaceKeyDestination),
+            $documentSource->getPath()
+        );
+
+        $this->output->writeln('<info>Processing: </info>' . $documentSource->getPath() . ' => ' . $newPath);
+
+        /** @var BasePageDocument $documentDestination */
+        try {
+            $documentDestination = $this->documentManager->find($newPath, $localeDestination);
+        } catch (\Exception $exception) {
+            $documentDestination = $this->documentManager->create('page');
+        }
+
+        // Set data.
+        $documentDestination->setTitle($documentSource->getTitle());
+        $documentDestination->setStructureType($documentSource->getStructureType());
+        $documentDestination->setWorkflowStage(WorkflowStage::TEST);
+        $documentDestination->setExtensionsData($documentSource->getExtensionsData());
+        $documentDestination->setResourceSegment($documentSource->getResourceSegment());
+        $documentDestination->setPermissions($documentSource->getPermissions());
+        $documentDestination->setSuluOrder($documentSource->getSuluOrder());
+        $documentDestination->setNavigationContexts($documentSource->getNavigationContexts());
+        $documentDestination->setShadowLocaleEnabled($documentSource->isShadowLocaleEnabled());
+        $documentDestination->setShadowLocale($documentSource->getShadowLocale());
```

Comment:
Just an idea: Would cloning the document work? Then we could only call the setters for the properties that really change...